torch/_dynamo/symbolic_convert.py

```diff
@@ -1060,11 +1060,10 @@ def POP_FINALLY(self, inst):
 
     def FOR_ITER(self, inst):
         it = self.pop()
-        if isinstance(it, ListIteratorVariable):
+        if isinstance(it, (variables.ListIteratorVariable, variables.IteratorVariable)):
             self.output.guards.update(it.guards)
             try:
-                val, next_iter = it.next_variables()
-                self.replace_all(it, next_iter)
```

Comment:
dropping this is very suspect

Reply:
we are just moving the ref to tx into `next_variable`, we then do `replace_all` inside of it. We need this to call `replace_all` on the inner iterator of cycle iterator